csharp/ql/test/library-tests/csharp9/TargetType.expected

```diff
@@ -0,0 +1,2 @@
+| TargetType.cs:12:18:14:18 | ... ? ... : ... | int? | int? | null |
```

Comment:
This means that the extractor does not insert implicit casts in the branches, I think it should. What do you think?

Reply:
I don't think we need the implicit casts, because that's not what happens in IL:
```
int? x = condition ? 5 : null;
```
becomes the following in IL:
```
Nullable<int> num = condition ? new Nullable<int>(5) : new Nullable<int>();
```

So there's no implicit cast. The problem is more like that we're not extracting nullables in the most accurate way. This could probably become part of https://github.com/github/codeql-csharp-team/issues/82.

Does it cause any issue in QL if the two branches of the conditional have different types? If so, then the short term fix is probably what you suggest, to add the implicit cast.

Reply:
OK; nullables are special (I forgot). But how about the other test case below?

Reply:
There's no `conv.` or `castclass` in IL for the latter case either, so why would we add a cast?
On the other hand, I think we add an implicit cast for the extraction of `IEnumerable<int> xs = new int[]{};`. And if we really do, then I think we should/must also add the cast for `?:` too.

What is the motivation behind having the implicit cast in the first place? 
What do you think?

Reply:
I am not convinced that the implicit casts are in fact that important. However, in cases where the implicit cast is via an operator, we should add it (both in assignments and in ternary conditionals).

Reply:
@hvitved I pushed an extra commit with some additional tests. Implicit casts and conversion calls do show up in the output.